integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/association/bytype/AbstractAutomaticIndexingAssociationBaseIT.java

```diff
@@ -450,6 +464,15 @@ public void directAssociationUpdate_indexedEmbedded() {
 
 			TContained oldContained = containingAssociation.get( entity1 );
 			containedAssociation.clear( oldContained );
+			containingAssociation.clear( entity1 );
+
+			if ( !isOutboxPollingCoordinationStrategyPresent() ) {
+				backendMock.expectWorks( _indexed().indexName() )
+						.addOrUpdate( "1", b -> {
+						} );
+			}
+			session.flush();
```

Comment:
I know the existing code is not very clear, but I have a suggestion to avoid complexifying it even more: could you group your additions together, with a comment on top explaining this is just to avoid problems with DB constraints?

You will need to do the same for every addition in other methods/files, though.

Something like this:

```suggestion

			// Clear the association first, to avoid problems with unique key constraints.
			containingAssociation.clear( entity1 );
			if ( !isOutboxPollingCoordinationStrategyPresent() ) {
				backendMock.expectWorks( _indexed().indexName() )
						.addOrUpdate( "1", b -> {
						} );
			}
			session.flush();
```

Or rather, with the condition I mentioned elsewhere:

```suggestion

			if ( !isAssociationMultiValuedOnContainingSide() ) {
				// Clear the association first, to avoid problems with unique key constraints.
				containingAssociation.clear( entity1 );
				if ( !isOutboxPollingCoordinationStrategyPresent() ) {
					backendMock.expectWorks( _indexed().indexName() )
							.addOrUpdate( "1", b -> {
							} );
				}
				session.flush();
			}
```